### PR TITLE
HBASE-23864 No need to submit SplitTableRegionProcedure/MergeTableReg…

### PR DESCRIPTION
**Pull request by infraio**
…ionsProcedure when split/merge is disabled

### CODE REVIEW

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/MergeTableRegionsProcedure.java [hunk 1]
```diff
@@ -453,13 +453,6 @@ private boolean prepareMergeRegion(final MasterProcedureEnv env) throws IOExcept
       throw new MergeRegionException("Skip merging regions " +
           RegionInfo.getShortNameToLog(regionsToMerge) + ", because we are snapshotting " + tn);
     }
-    if (!env.getMasterServices().isSplitOrMergeEnabled(MasterSwitchType.MERGE)) {
```
Comment (Apache9) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/MergeTableRegionsProcedure.java]: You can add extra checks before scheduling the procedure to make the request fail fast, but I do not think we should remove the checks in the procedure? There could be race, that when scheduling, split and merge are enabled but when you actually execute the procedure, the flags are turned to disabled...

Comment (binlijin) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/MergeTableRegionsProcedure.java]: Agree with Duo, others are good.

Comment (infraio) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/MergeTableRegionsProcedure.java]: I thought the race is not a big deal? And our admin client never have this guarantee: interrupt the schduled Split/Merge after split/merge is disabled.

Comment (Apache9) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/MergeTableRegionsProcedure.java]: But split and merge can be scheduled by background task? And also, if you remove the checks here, then you need to add the check everywhere...

Comment (infraio) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/MergeTableRegionsProcedure.java]: Yes. I checked this. Now only two place can schedule split/merge task. One is the rpc request of master. Another one is from RS's report.

Comment (Apache9) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/MergeTableRegionsProcedure.java]: But if in the future we schedule these procedures in other places, we need to add the check there...

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 11s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 59s |  master passed  |
| +1 :green_heart: |  compile  |   1m  1s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 18s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m  7s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 38s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 34s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 32s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 47s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  2s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  2s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 15s |  hbase-server: The patch generated 2 new + 116 unchanged - 0 fixed = 118 total (was 116)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   5m  7s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  17m 36s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 38s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   5m  8s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 101m 24s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 26s |  The patch does not generate ASF License warnings.  |
|  |   | 165m 26s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.6 Server=19.03.6 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1182 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux ba7df4c35510 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1182/out/precommit/personality/provided.sh |
| git revision | master / c4fb1cd234 |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/1/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/1/testReport/ |
| Max. process+thread count | 6800 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 17s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 58s |  master passed  |
| +1 :green_heart: |  compile  |   1m  0s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 17s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m  3s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 39s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 32s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 30s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 15s |  the patch passed  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 9 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  shadedjars  |   5m  3s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  18m  8s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 38s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 37s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 105m 48s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 43s |  The patch does not generate ASF License warnings.  |
|  |   | 169m 30s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.6 Server=19.03.6 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1182 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 66885089d599 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1182/out/precommit/personality/provided.sh |
| git revision | master / c4fb1cd234 |
| Default Java | 1.8.0_181 |
| whitespace | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/2/artifact/out/whitespace-eol.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/2/testReport/ |
| Max. process+thread count | 6628 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   4m 28s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m  1s |  master passed  |
| +1 :green_heart: |  compile  |   1m  1s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 17s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m  7s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 49s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 47s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 40s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  1s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  1s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 17s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   5m  8s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  17m 38s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   1m 21s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   6m 51s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  97m  6s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 25s |  The patch does not generate ASF License warnings.  |
|  |   | 167m 15s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.6 Server=19.03.6 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1182 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 799dccd0d6be 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1182/out/precommit/personality/provided.sh |
| git revision | master / c4fb1cd234 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/3/testReport/ |
| Max. process+thread count | 6456 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 21s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 22s |  master passed  |
| +1 :green_heart: |  compile  |   1m  9s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 25s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 18s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 43s |  master passed  |
| +0 :ok: |  spotbugs  |   5m 33s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m 31s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 10s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 10s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 10s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 23s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   5m 34s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  18m 51s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   6m 59s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 143m 31s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 52s |  The patch does not generate ASF License warnings.  |
|  |   | 214m 49s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.replication.TestNamespaceReplicationWithBulkLoadedData |
|   | hadoop.hbase.replication.TestMasterReplication |
|   | hadoop.hbase.security.access.TestAccessController3 |
|   | hadoop.hbase.security.access.TestSnapshotScannerHDFSAclController |
|   | hadoop.hbase.replication.TestReplicationStatusAfterLagging |
|   | hadoop.hbase.replication.regionserver.TestReplicationSink |
|   | hadoop.hbase.replication.regionserver.TestRegionReplicaReplicationEndpointNoMaster |
|   | hadoop.hbase.replication.TestSyncReplicationStandbyKillRS |
|   | hadoop.hbase.replication.regionserver.TestWALEntryStream |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.6 Server=19.03.6 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1182 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux ebdb9156ab47 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1182/out/precommit/personality/provided.sh |
| git revision | master / 86e76c2bd9 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/4/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/4/testReport/ |
| Max. process+thread count | 6954 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1182/4/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by infraio** [comment]
The failed ut not related. Ping @Apache9 for reviewing.

**Pull request merged**